tests/bulk/bulkwrite-ctor-let-001.phpt

```diff
@@ -0,0 +1,68 @@
+--TEST--
+MongoDB\Driver\BulkWrite::__construct(): let option
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_server_version('<', '5.0'); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+
+require_once __DIR__ . "/../utils/basic.inc";
+
+class CommandLogger implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+    public function commandStarted(MongoDB\Driver\Monitoring\CommandStartedEvent $event)
+    {
+        $command = $event->getCommand();
+
+        if (!isset($command->let)) {
+            printf("%s does not include let option\n", $event->getCommandName());
+
+            return;
+        }
+
+        printf("%s included let: %s\n", $event->getCommandName(), json_encode($event->getCommand()->let));
+    }
+
+    public function commandSucceeded(MongoDB\Driver\Monitoring\CommandSucceededEvent $event)
+    {
+    }
+
+    public function commandFailed(MongoDB\Driver\Monitoring\CommandFailedEvent $event)
+    {
+    }
+}
+
+$manager = create_test_manager();
+
+$bulk = new MongoDB\Driver\BulkWrite(['let' => ['id' => 1, 'x' => 'foo']]);
+$bulk->insert(['_id' => 1]);
+$bulk->insert(['_id' => 2]);
+$bulk->delete(['$expr' => ['$eq' => ['$_id', '$$id']]]);
+$bulk->update(['_id' => 2], [['$set' => ['x' => '$$x']]]);
+
+$manager->addSubscriber(new CommandLogger);
+$manager->executeBulkWrite(NS, $bulk);
+
+$cursor = $manager->executeQuery(NS, new MongoDB\Driver\Query([]));
+var_dump($cursor->toArray());
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+insert does not include let option
+delete included let: {"id":1,"x":"foo"}
+update included let: {"id":1,"x":"foo"}
+find does not include let option
```

Comment:
As far as I get the idea, this test is designed for BulkWrite constructor and that's why it includes both delete and update operations. But may be in such case it would be more flexible to arrange constructor tests not for single argument?

Reply:
>  I meant tests arrangement looks like they a separated in groups of actions: delete, insert, update. So constructor tests (as you added prefix "ctor" for new test) can be arranged in separate group, which can verify not only "let" argument, but may be some more actions or arguments.

Following up on your Slack comment. The convention for naming tests is generally `class-method-###.phpt`, and there's some variation for `_error` tests as you've seen. For things that don't have methods, we typically just fill in the name of the operation instead of "method", such as `debug` for testing the `get_debug_info` handler (via `var_dump()`) or `serialize` for testing how the object behaves with `serialize()`.

Since some of our constructors support _many_ options, it's not feasible to stick to `class-ctor-###.phpt` for all of them, as the filenames wouldn't be very useful and we'd just end up with lots and lots of tests. The tests for the Manager class are a good example of this where we have `manager-ctor-optionname-###.phpt` and that's what I've decided to borrow here. Using "ctor" is just personal preference since "__construct` would be more verbose and both convey the same information.

If/when we introduce a `bypassDocumentValidation` test, we can certainly do so in a new `bulkwrite-ctor-bypassdocumentvalidation-001.phpt` file. And seeing the existing `let` option test being added in this PR would likely serve as an example for the author to do so down the line.

I'll also point out that most of the actual method tests (e.g. `bulkwrite-update-*`) only use a numeric sequence, as demonstrated by the `BulkWrite::update()` test for `hint` I linked above. There typically aren't very many tests for individual methods are there are for constructors, which is why we've not bothered to create names like `bulkwrite-update-hint-001.phpt` -- although we certainly could. In the case of `let`, this was the very first `bulkwrite-ctor` test so I decided to adopt the style we employed for Manager.